### PR TITLE
Basic Github workflow when Pull Request is Created. 

### PR DESCRIPTION
**Pull request by Hrushi20**
Created a Github workflow which runs when a Pull request is created. Kept it simple. 

I've added a job to check the lint. We can improvise on this further to check for builds, validator setups etc.

### CODE REVIEW

File: .github/workflows/pull-requests.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          node-version: ${{ matrix.node }}
+          cache: 'npm'
+
+      - name: Install dependencies
+        shell: bash
+        run: npm install
+      
+      - name: Linting
+        shell: bash
+        run: npm run format
```
Comment (buffalojoec) [inline .github/workflows/pull-requests.yaml]: I have no idea if this is going to work, but we can get even simpler:

```suggestion
jobs:
  check-formatting:

    steps:
      - name: Checkout
         uses: actions/checkout@v4

      - name: Install Dependencies
         cache: 'npm'
         run: npm install
         shell: bash

      - name: Test Lint
        run: npm run lint
        shell: bash

      - name: Test Prettier
        run: npm run prettier
        shell: bash
```

**Review by buffalojoec — COMMENTED** [review]
Thanks! We definitely need this.

**Comment by Hrushi20** [comment]
I've tested it out. It works. I'll update it with simpler workflow file.

**Comment by Hrushi20** [comment]
Squashed my commits.

**Review by buffalojoec — APPROVED** [review]
Thanks so much!

**Pull request merged**